src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11AEADCipher.java

```diff
@@ -356,9 +356,13 @@ private void implInit(int opmode, Key key, byte[] iv, int tagLen,
                 encrypt = false;
                 requireReinit = false;
                 break;
-            default:
-                throw new InvalidAlgorithmParameterException
+            case Cipher.WRAP_MODE:
+            case Cipher.UNWRAP_MODE:
+                throw new UnsupportedOperationException
                         ("Unsupported mode: " + opmode);
+            default:
+                // should never happen; checked by Cipher.init()
+                throw new AssertionError("Unknown mode: " + opmode);
```

Comment:
This update looks good to me.  But there is a potential issues, maybe for all switch statements.  If the cipher modes is extended to support more items, we may have to look for and update all the switch uses and make sure there is no broken.  Anyway, for my personal understanding, there is nothing we could do right now.

Reply:
Right, if new modes are added, we need to go through all cipher impl and update the switch statements. This change is just to make sure our handling is consistent.

Reply:
Thanks much for the review~